zio-http/src/test/scala/zhttp/endpoint/EndpointSpec.scala

```diff
@@ -61,6 +62,18 @@ object EndpointSpec extends DefaultRunnableSpec {
         assert(route.extract(!! / "1" / "c"))(isSome(equalTo(1))) &&
         assert(route.extract(!! / "1"))(isNone) &&
         assert(route.extract(!! / "c"))(isNone)
+      } +
+      test("combining endpoints should resolve second") {
+        val a      = Method.GET / "a" / *[Int] to { _ => Response.ok }
+        val b      = Method.GET / "b" / *[Int] to { _ => Response.ok }
+        val actual = (a ++ b).execute(Request(Method.GET, URL(Path("/b/2")))).map(_.status)
+        assert(actual)(equalTo(HExit.succeed(Status.OK)))
+      } +
+      testM("combining endpoints should resolve second effect") {
+        val a      = Method.GET / "a" / *[Int] to { _ => UIO(Response.ok) }
+        val b      = Method.GET / "b" / *[Int] to { _ => UIO(Response.ok) }
+        val actual = (a ++ b).execute(Request(Method.GET, URL(Path("/b/2")))).map(_.status)
+        assertM(actual.toEffect)(equalTo(Status.OK))
```

Comment:
Write tests for `route` not `Http` here.

Reply:
@tusharmath  I have added tests based on Http as it is about testing combining end points which results in an Http, like this:
```
val app: Http[Any, Nothing, Request, Response[Any, Nothing]] = h2 ++ h1
```
Should we stick to Http based assertions then?